kotest-assertions/kotest-assertions-shared/src/jvmMain/kotlin/io/kotest/data/TableFile.kt

```diff
@@ -0,0 +1,71 @@
+package io.kotest.data
+
+import io.kotest.matchers.shouldBe
+import java.io.File
+
+// TODO I'm only supporting table with 3 arguments until the API stabilizes
+
+fun <A, B, C> table(
+   headers: Headers3,
+   file: File,
+   transform: (String, String, String) -> Row3<A, B, C>
+): Table3<A, B, C> {
+   val rows = file.readStringTable(headers.values())
+      .mapRows { (a, b, c) -> transform(a, b, c) }
+   return Table3(headers, rows)
+}
+
+fun File.writeTable(headers: Headers1, rows: List<Row1<String>>): String =
+   writeTable(headers.values(), rows.map { it.strings() } )
+fun File.writeTable(headers: Headers2, rows: List<Row2<String, String>>): String =
+   writeTable(headers.values(), rows.map { it.strings() } )
+fun File.writeTable(headers: Headers3, rows: List<Row3<String, String, String>>): String =
+   writeTable(headers.values(), rows.map { it.strings() } )
+// TODO
+
+private fun Row.strings(): List<String> = values().map { it.toString() }
+
+
+internal fun File.readStringTable(headers: List<String>): StringTable {
+   if (exists().not()) throw AssertionError("Can't read table file")
+   if (extension != "table") throw AssertionError("Table file must have a .table extension")
+   val lines = readLines()
+   if (lines.isEmpty()) throw AssertionError("Table file must have a header")
```

Comment:
Would it make sense to throw `IllegalArgumentException`s instead? No assertion has been made at this point, I think?

Reply:
you can decide. Personally I don't like exceptions since they don't work well with `assertSoftly` for example.

```kotlin
   test("invalid file") {
      val file = resourcesDir.resolve("invalid.table") 
      val table = table(headers, file, transform) // implicit assertion that the file exists
      table shouldBe expectedTable
   }
```

Reply:
I think there's a difference between an assertion (E.g. validating the output of production code) and a bug in test code. IMO this `AssertionError` would indicate a fault in the test code, not the main code and I think the distinction is clearer when output as an exception. What do you think?